src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -108,7 +112,9 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 				? this.messageKeyExpression.getValue(this.evaluationContext, message)
 				: message.getHeaders().get(KafkaHeaders.MESSAGE_KEY);
 
-		this.kafkaProducerContext.send(topic, partitionId, messageKey, message.getPayload());
+		Future<RecordMetadata> future = this.kafkaProducerContext.send(topic, partitionId, messageKey,
```

Comment:
Thanks @artembilan!

The `KafkaTemplate` comment was rather speculative at this point, I think that the main reason why this change felt necessary was that the synchronous option is not very obvious to the `KafkaProducerMessageHandler` user, and may need to be made available at a more abstract level (e.g. part of the `sendCall`, maybe have an option to `sendSynchronously`), but that is more concerning future design. Sorry for sidetracking :).